### PR TITLE
Migrate to AVKit

### PR DESCRIPTION
**Pull request by j-maas**
`TODO`s mark places where I'm unsure about the migration.

The major todos are:
- [x] Reintroduce file type checking.
- [ ] Reintroduce exporting.
- [ ] Write temp video async.
- [ ] Fix frame generation on mouse drag. (I don't know precisely what the frame is for.)

The first commit to compile again is b1933cfa8b33740914cb1b598bfb7403b628a394.

### CODE REVIEW

File: Classes/Core/AppController.m [hunk 1]
```diff
@@ -3496,9 +3371,9 @@ - (void)updateDisplay:(NSTimer *)aTimer
         
 		if(annotationPlayback && selectedAnnotation && [selectedAnnotation isDuration])
 		{
-            QTTime currentTime = [mMovie currentTime];
-            QTTime normalizedStart = QTMakeTimeScaled([selectedAnnotation startTime], currentTime.timeScale);
-			if(QTTimeCompare(currentTime,[selectedAnnotation endTime]) == NSOrderedDescending)
+            CMTime currentTime = [mMovie currentTime];
+            CMTime normalizedStart = CMTimeConvertScale([selectedAnnotation startTime], currentTime.timescale, kCMTimeRoundingMethod_Default); // TODO: Check if QTMakeTimeScaled is correctly replacesd by CMTimeConvertScale
```
Comment (j-maas) [inline Classes/Core/AppController.m:3397]: Is this replacement correct? I can't find any documentation on `QTMakeTimeScaled` to ensure that it did the same as `CMTimeConvertScale` does.

Comment (j-maas) [reply Classes/Core/AppController.m:3397]: Note that there is the [same todo in `TranscriptView.m`](https://github.com/afouse/chronoviz/blob/avkit-attempt/Classes/Views/TranscriptView.m#L114).

Comment (afouse) [reply Classes/Core/AppController.m:3397]: @Y0hy0h I looked into this a bit, and this is the correct replacement.

**Comment by j-maas** [comment]
It looks like this is already merged into `avkit`.

**Pull request closed without merging**